SWIPs/swip-20.md

```diff
@@ -8,3 +8,118 @@ type: Standards Track
 category: Core
 created: 2024-05-08
 ---
+
+
+## Abstract
+
+Currently, one's initial capital, invested as stake, is permanently locked. This means that returns on this stake is purely based on the ROI of the redistribution game. The purpose of this SWIP is to change that: in extenuating circumstances, users will be allowed to withdraw their stakes.
+
+
+## Objectives
+
+Should the effective price of a user's stake increase, they should be able to withdraw the appreciation, i.e., the amount that is no longer representative of the capital required to play and safeguard the system.
+
+
+## Context
+
+Currently, the smart contract can only increase a node's stake by sending extra collateral. Importantly, it does not allow node operators to  decrease their stake or any other way fully or partially withdraw their stake. This is what will be changed with this feature.
+
+
+## Specification
+
+This part describes how stakes should be created, modified, computed and withdrawn. As a first step, we need to obtain the *committed stake* of the user. This is set by user action, based on the following procedure:
+
+### The stake endpoint
```

Comment:
this endpoint naming is confusing for me. would it be the currently existing `depositStake` function? or the Staking contract itself?

Reply:
ú-yes I mean the depositStake function. But talking from a bee perspective this is used as the backend of the `/stake` Bee Api.